[jason-raiin] iP

jason-raiin opened the pull request:
## DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

jason-raiin committed:
Add .vscode folder to .gitignore

jason-raiin committed:
Rename chatbot

jason-raiin committed:
Add greet and exit messages

jason-raiin committed:
Initialise scanner

jason-raiin committed:
Implement echo functionality

jason-raiin committed:
Add Command enum and parse method

jason-raiin committed:
Implement switching for different commands

jason-raiin committed:
Initialise list, implement add to list, implement list command

jason-raiin committed:
Add Task class

jason-raiin committed:
Standardise formatting

jason-raiin committed:
Implement Task class in chatbot

jason-raiin committed:
Add input processing

jason-raiin committed:
Implement mark and unmark commands

jason-raiin committed:
Add todo, deadline, and event commands

jason-raiin committed:
Add Event, Task, and Todo classes, made Task abstract

jason-raiin committed:
Restore default Echo functionality

jason-raiin committed:
Override toString method for Task subclasses

jason-raiin committed:
Implement Deadline and Event

jason-raiin committed:
Fix minor UI bugs, implement basic testing for Jerma

jason-raiin committed:
Add exception throw and catch for invalid command

jason-raiin committed:
Streamline exception handling

jason-raiin committed:
Implement Delete command

jason-raiin committed:
Add save method to Task to generate save string

jason-raiin committed:
Add SaveHandler class and implement save method

jason-raiin committed:
Implement load method

jason-raiin committed:
Add logging for loading and saving

jason-raiin committed:
Handle filepaths independent of OS

jason-raiin committed:
Handle directory creation and IOExceptions

jason-raiin committed:
Merge branch 'branch-Level-7'

jason-raiin committed:
Add LocalDate parsing and exception handling

jason-raiin committed:
Merge branch 'branch-Level-8'

jason-raiin committed:
Move all tasks into tasks package

jason-raiin committed:
Rename SaveHandler to Storage

jason-raiin committed:
Refactor out TaskList and UI

jason-raiin committed:
Move utils classes into utils package

jason-raiin committed:
Add new Command abstract class and implement List class

jason-raiin committed:
Refactor Mark and Unmark commands

jason-raiin committed:
Add Delete command

jason-raiin committed:
Add Bye command

jason-raiin committed:
Improve error handling

jason-raiin committed:
Add AddTask command

jason-raiin committed:
Refactor input parsing into Parser utils class

jason-raiin committed:
Move error handling to main file

jason-raiin committed:
Move scanner to Ui

jason-raiin committed:
Merge branch 'A-MoreOOP'

jason-raiin committed:
Package all files into Jerma package

jason-raiin committed:
Merge branch 'A-Packages'

jason-raiin committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

jason-raiin committed:
Add checkstyles

jason-raiin committed:
Configure shadowJar

jason-raiin committed:
Merge branch 'A-Gradle'

jason-raiin committed:
Add unit tests for tasks

jason-raiin committed:
Merge branch 'A-JUnit'

jason-raiin committed:
Add javadocs for Tasks

jason-raiin committed:
Add Find command

jason-raiin committed:
Merge branch 'branch-A-Javadoc'

jason-raiin committed:
Merge branch 'branch-Level-9'

jason-raiin committed:
Add javafx dependencies

jason-raiin committed:
Add images for user and bot

jason-raiin committed:
Implement basic echo functionality for GUI

jason-raiin committed:
Integrate with chatbot

jason-raiin committed:
Integrate fxml

jason-raiin committed:
Add on start events

jason-raiin committed:
Merge branch 'branch-Level-10'

keaganpzh commented on src/main/java/jerma/commands/AddTask.java:
```diff
@@ -0,0 +1,19 @@
+import jerma.utils.TaskList;
+import jerma.utils.Ui;
+
+public class AddTask extends Command {
```
Missing header comments for your classes

keaganpzh commented on src/main/java/jerma/commands/AddTask.java:
```diff
@@ -0,0 +1,19 @@
+public class AddTask extends Command {
+    private Task task;
+
+    public AddTask(Ui ui, TaskList tasks, Task task) {
```
Some methods are missing header comments too

keaganpzh commented on src/main/java/jerma/utils/Parser.java:
```diff
@@ -0,0 +1,75 @@
+import jerma.tasks.Task;
+import jerma.tasks.Todo;
+
+enum Keywords {
```
Nice use of enums!

keaganpzh commented on src/main/java/jerma/utils/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        String[] inputArgs = input.split(" ", 2);
+        Keywords command = Keywords.parse(inputArgs[0]);
+
+        switch (command) {
```
Good that you caught the rule on switch statement indentations

keaganpzh commented on src/main/resources/images/jerma.jpg:
sus guy +100000

keaganpzh commented on src/test/java/jerma/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,41 @@
+package jerma.tasks; //same package as the class being tested
+
+import org.junit.jupiter.api.Test;
```
The import order might be abit wrong here

keaganpzh reviewed (commented):
Good job man your code looks pretty clean. Keep it up!!

Kb-Tay commented on src/main/java/jerma/utils/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        }
+
+        throw new UnsupportedOperationException();
+    }
```
Good use of switch case which ensures readability of the code. Could you have used default case for unknown commands?

Kb-Tay commented on src/main/java/jerma/commands/AddTask.java:
```diff
@@ -0,0 +1,19 @@
+        this.task = task;
+    }
+
+    public String execute() {
+        this.tasks.add(this.task);
+        return this.ui.newTask(this.task);
+    }
```
Good use of abstraction for the execute function for the command classes.

Kb-Tay commented on src/main/java/jerma/commands/Find.java:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    public String execute() {
+        TaskList results = new TaskList();
+        for (Task task : tasks) {
+            if (task.toString().contains(keyword)) {
+                results.add(task);
+            }
+        }
```
Could you have abstracted this into the TaskList class rather than getting the task then adding to a TaskList?

Kb-Tay commented on src/test/java/jerma/tasks/EventTest.java:
```diff
@@ -0,0 +1,44 @@
+import java.time.format.DateTimeParseException;
+
+public class EventTest {
+    private final Task event = new Event("desc", "2019-08-09", "2019-09-09");
```
Could possibly used constants to represent the dates to avoid repetition.

jason-raiin committed:
Add help method to Ui

jason-raiin committed:
Add Help command

jason-raiin committed:
Add Help command to parser

jason-raiin committed:
Merge branch 'branch-C-Help'

jason-raiin committed:
Add assertions for index argument

MARK UNMARK and DELETE commands accept an index argument for the list
item to select. Assertion added to check that the index provided is
within the range of indexes in the list.

jason-raiin committed:
Remove redundant this references

jason-raiin committed:
Merge pull request #1 from jason-raiin/branch-A-CodeQuality

Remove redundant this references

jason-raiin committed:
Merge branch 'master' of https://github.com/jason-raiin/ip

jason-raiin committed:
Merge branch 'master' into branch-A-Assertions

jason-raiin committed:
Merge pull request #2 from jason-raiin/branch-A-Assertions

Add assertions for list index

jason-raiin committed:
Merge branch 'master' of https://github.com/jason-raiin/ip

jason-raiin committed:
Add CI yml file

jason-raiin committed:
Merge pull request #3 from jason-raiin/A-CI

Add CI

jason-raiin committed:
Merge branch 'master' of https://github.com/jason-raiin/ip

ceilingFan456 commented on src/main/java/jerma/utils/Storage.java:
```diff
@@ -0,0 +1,67 @@
+            String[] taskElements = taskString.split("\\|");
+            Task task;
+
+            switch (taskElements[0]) {
```
I like the nice switch statement format here.

ceilingFan456 commented on src/main/java/jerma/utils/Ui.java:
```diff
@@ -0,0 +1,62 @@
+
+    public String deleteTask(Task task, int numberOfTasksRemaining) {
+        String output = String.format(
+                "Removed the task: \n%s \nYou have %d tasks remaining.", task,
```
the line wrapping here is really clear following the 8-space rule.

ceilingFan456 commented on src/main/resources/images/jerma.jpg:
lol I was jump-scared by this. good choice of photo hahah.

ceilingFan456 commented on src/main/java/jerma/commands/Delete.java:
```diff
@@ -0,0 +1,20 @@
+
+    public Delete(Ui ui, TaskList tasks, int index) {
+        super(ui, tasks);
+        assert index > 0 || index <= tasks.size() : "Index out of range";
```
should this be handled by exception instead of assertion?

ceilingFan456 commented on src/main/java/jerma/Jerma.java:
```diff
@@ -0,0 +1,53 @@
+public class Jerma {
+    private TaskList tasks;
+    private Ui ui;
+    private Boolean[] running;
```
Should the name be modified so that it is a better representation of a boolean array?

ceilingFan456 commented on src/main/java/jerma/utils/Storage.java:
```diff
@@ -0,0 +1,67 @@
+        }
+
+        String saveString = "";
+        for (Task task : tasks) {
```
nice following the format for for loop although there is only one line within the loop.

ceilingFan456 reviewed (commented):
Really nice job on following all the coding standards. your code looks clear and clean to read.

jason-raiin committed:
Implement quitting the programme on bye command

jason-raiin committed:
Fix all checkstyle errors

jason-raiin committed:
Fix checkstyle for tests

jason-raiin committed:
Remove text ui tests

jason-raiin committed:
Fill in README docs

jason-raiin committed:
Add sample UI image
